src/WinXP/apps/Notepad/index.js

```diff
@@ -99,24 +162,129 @@ export default function Notepad({ onClose }) {
     }
   }
 
+  const updateSettings = newSettings => {
+    setFindSettings({ ...findSettings, ...newSettings });
+  };
+
+  const findNext = newSettings => {
+    let settings = findSettings;
+
+    if (newSettings) {
+      setFindSettings(newSettings);
+      settings = { ...settings, ...newSettings };
+    }
+    /// Conduct the search
+
+    const index = getIndex(settings);
+    if (index !== -1) selectText(index, index + settings.searchWord.length);
+    else
+      openApp(
+        'InfoDialog',
+        { info: `Cannot find "${settings.searchWord}"` },
+        {
+          header: { ...appSettings.InfoDialog.header, title: 'Notepad' },
+        },
+      );
+  };
+
+  const getIndex = ({ forwardSearch, searchWord, caseSensitive }) => {
+    let searchStr = textareaRef.current.value;
+
+    if (!caseSensitive) {
+      searchStr = docText.toLowerCase();
+      searchWord = searchWord.toLowerCase();
+    }
+
+    /// Reaching index 0 in 'up' search
+    if (!forwardSearch && caretStart.current === 0) return -1;
+
+    return forwardSearch
+      ? searchStr.indexOf(searchWord, caretEnd.current)
+      : searchStr.lastIndexOf(searchWord, caretStart.current - 1);
+  };
+
+  function openApp(app, props, customAppSettings) {
+    appContext.dispatch({
+      type: ADD_APP,
+      payload: {
+        ...appSettings[app],
+        injectProps: props,
+        customAppSettings,
+      },
+    });
+    /// Reselect text
+    selectText(caretStart.current, caretEnd.current);
+  }
+
+  function onOpenFind() {
+    openApp('FindDialog', { findSettings, findNext, updateSettings });
+  }
+
+  function onOpenReplace() {
+    openApp('ReplaceDialog', {
+      findSettings,
+      findNext,
+      onReplace,
+      onReplaceAll,
+      updateSettings,
+    });
+  }
+
+  const frameInitialContent = `
+  <!DOCTYPE html><html>
+  <head>
+  <style>
+  html, div, body, textarea {
+    height: 100%;
+    width: 100%;
+    margin: 0;
+  }
+  html, body {
+    overflow: hidden;
+  }
+  textarea {
+    flex: auto;
+    outline: none;
+    font-family: 'Lucida Console', monospace;
+    font-size: 13px;
+    line-height: 14px;
+    resize: none;
+    padding: 2px;
+    ${wordWrap ? '' : 'white-space: nowrap; overflow-x: scroll;'}
+    overflow-y: scroll;
+    border: 1px solid #96abff;
+  }
+
+  textarea::selection {
+    background-color: #1660e8;
+    color: white;
+  }
+  </style>
+  </head>
+  <body><div></div></body></html>`;
```

Comment:
Do we really need this extra `div`?

Reply:
@RonnyO It seems like it is required for the frameInitialContent to have a DOM element inside the body tag, else it throws an error "Target container is not a DOM element". I can dig deeper into it and see if I can find a workaround

Reply:
Found in the documentation of the react-frame-component under `initialContent` : "Also at least one div is required in the body of the html, which we use to render the react dom into."